lib/uloop.c

```diff
@@ -948,11 +924,17 @@ uc_uloop_process_delete(uc_vm_t *vm, size_t nargs)
 static void
 uc_uloop_process_cb(struct uloop_process *proc, int exitcode)
 {
-	uc_uloop_process_t *process = (uc_uloop_process_t *)proc;
+	uc_uloop_process_t *process = container_of(proc, uc_uloop_process_t, process);
 	uc_value_t *e = ucv_int64_new(exitcode >> 8);
+	void **dataptr;
+
+	uc_uloop_cb_invoke(&process->cb, e);
 
-	uc_uloop_reg_invoke(process->vm, process->registry_index, e);
-	uc_uloop_process_clear(&process);
+	dataptr = ucv_resource_dataptr(process->cb.obj, "uloop.process");
+	if (dataptr)
+		*dataptr = NULL;
```

Comment:
I think this code will never run since `uc_uloop_process_t` is allocated as extended resource as well.